dm/pkg/checker/binlog.go

```diff
@@ -262,3 +273,109 @@ func (c *BinlogDBChecker) Check(ctx context.Context) *Result {
 func (c *BinlogDBChecker) Name() string {
 	return "binlog_do_db/binlog_ignore_db check"
 }
+
+// MetaPositionChecker checks if meta position for given source database is valid
+type MetaPositionChecker struct {
+	db         *conn.BaseDB
+	sourceCfg  dbconfig.DBConfig
+	enableGTID bool
+	meta       *config.Meta
+}
+
+// NewBinlogDBChecker returns a RealChecker.
+func NewMetaPositionChecker(db *conn.BaseDB, sourceCfg dbconfig.DBConfig, enableGTID bool, meta *config.Meta) RealChecker {
+	return &MetaPositionChecker{db: db, sourceCfg: sourceCfg, enableGTID: enableGTID, meta: meta}
+}
+
+// Check implements the RealChecker interface.
+func (c *MetaPositionChecker) Check(ctx context.Context) *Result {
+	result := &Result{
+		Name:  c.Name(),
+		Desc:  "check whether meta position is valid for db",
+		State: StateFailure,
+		Extra: fmt.Sprintf("address of db instance - %s:%d", c.sourceCfg.Host, c.sourceCfg.Port),
+	}
+
+	var tlsConfig *tls.Config
+	var err error
+	if c.sourceCfg.Security != nil {
+		if loadErr := c.sourceCfg.Security.LoadTLSContent(); loadErr != nil {
+			markCheckError(result, loadErr)
+			return result
+		}
+		tlsConfig, err = util.NewTLSConfig(
+			util.WithCAContent(c.sourceCfg.Security.SSLCABytes),
+			util.WithCertAndKeyContent(c.sourceCfg.Security.SSLCertBytes, c.sourceCfg.Security.SSLKeyBytes),
+			util.WithVerifyCommonName(c.sourceCfg.Security.CertAllowedCN),
+		)
+		if err != nil {
+			markCheckError(result, err)
+			return result
+		}
+	}
+
+	flavor, err := conn.GetFlavor(ctx, c.db)
+	if err != nil {
+		markCheckError(result, err)
+		return result
+	}
+
+	// always use a new random serverID
+	randomServerID, err := conn.GetRandomServerID(tcontext.NewContext(ctx, log.L()), c.db)
+	if err != nil {
+		// should never happened unless the master has too many slave
+		markCheckError(result, terror.Annotate(err, "fail to get random server id for relay reader"))
+		return result
+	}
+
+	h, _ := os.Hostname()
+	h = "dm-checker-" + h
+	// https://github.com/mysql/mysql-server/blob/1bfe02bdad6604d54913c62614bde57a055c8332/include/my_hostname.h#L33-L42
+	if len(h) > 60 {
+		h = h[:60]
+	}
+
+	syncCfg := replication.BinlogSyncerConfig{
+		ServerID:  randomServerID,
+		Flavor:    flavor,
+		Host:      c.sourceCfg.Host,
+		Port:      uint16(c.sourceCfg.Port),
+		User:      c.sourceCfg.User,
+		Password:  c.sourceCfg.Password,
+		TLSConfig: tlsConfig,
+		Localhost: h,
+	}
+
+	syncer := replication.NewBinlogSyncer(syncCfg)
```

Comment:
need to close syncer

Reply:
fixed in https://github.com/pingcap/tiflow/pull/9000/commits/73fdfe365c13389a08aab07a7452cd475d64ad15